src/main/java/lotto/view/UserIoManager.java

```diff
@@ -0,0 +1,72 @@
+package lotto.view;
+
+import lotto.model.lotto.Bonus;
+import lotto.model.lotto.Lotto;
+import lotto.model.lotto.WinningNumbers;
+import lotto.model.user.UserLotto;
+import lotto.model.user.UserLottoRank;
+import lotto.model.user.UserMoney;
+
+public class UserIoManager {
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public UserIoManager(InputView inputView, OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    public UserMoney readUserMoney() {
+        while (true) {
+            try {
+                outputView.printUserMoneyInputGuide();
+                int userMoney = inputView.readUserMoney();
+                return new UserMoney(userMoney);
+            } catch (IllegalArgumentException e) {
+                printErrorMessage(e.getMessage());
+            }
+        }
```

Comment:
이 부분은 뎁스를 1로 유지하도록 리팩토링 해볼 수 있을 것 같아요!

Reply:
예외가 발생했을 때 메서드를 다시 호출함으로써 무한반복을 유도하라는 뜻이군요!
생각치도 못한 아이디어라 덕분에 영감을 얻었습니다🫢

```suggestion
        try {
            outputView.printUserMoneyInputGuide();
            int userMoney = inputView.readUserMoney();
            return new UserMoney(userMoney);
        } catch (IllegalArgumentException e) {
            printErrorMessage(e.getMessage());
            return readUserMoney();
        }
```